plugins/Jobs.cpp

```diff
@@ -161,13 +187,43 @@ bool BedrockPlugin_Jobs::peekCommand(SQLite& db, BedrockCommand& command) {
     
     // ----------------------------------------------------------------------
     else if (SIEquals(request.methodLine, "CreateJob")) {
+        if (request.isSet("repeat") && request.isSet("retryAfter")) {
+            throw "402 Repeatable auto-retrying jobs are not supported";
+        }
+
+        // If parentJobID is passed, verify that the parent job doesn't have a retryAfter set
+        int64_t parentJobID = request.calc64("parentJobID");
+        if (parentJobID) {
+            SINFO("parentJobID passed, checking existing job with ID " << parentJobID);
+            SQResult result;
+            if (!db.read("SELECT retryAfter "
+                        "FROM jobs "
+                        "WHERE jobID=" + SQ(parentJobID) + ";",
+                        result)) {
+                throw "502 Select failed";
+            }
+            if (!result.empty() && result[0][0] != "") {
+                throw "402 Auto-retrying parents cannot have children";
+            }
+        }
+
+        // Validate retryAfter
+        if (request.isSet("retryAfter") && request["retryAfter"] != "" && !_isValidSQLiteDateModifier(request["retryAfter"])){
+            throw "402 Malformed retryAfter";
+        }
+
         // If unique flag was passed and the job exist in the DB, then we can
         // finish the command without escalating to master.
         if (!request.test("unique")) {
             // Not unique; need to process
             return false;
         }
 
+        // Throw if retryAfter was passed for unique job
+        if (request.isSet("retryAfter")) {
+            throw "405 Unique jobs can't be retried";
```

Comment:
@quinthar I did this in the spirit of being safe since we haven't defined the behavior for auto-retrying unique jobs. Thoughts?